.circleci/config.yml

```diff
@@ -1386,6 +1387,175 @@ jobs:
+                wget -O .\python_installer\python-3.7.7-amd64.exe https://www.python.org/ftp/python/3.7.7/python-3.7.7-amd64.exe
+            }
+            .\python_installer\python-3.7.7-amd64.exe /install /quiet TargetDir=c:\Python37 Include_tcltk=0
+
+      - save_cache:
+          key: windows_python_installer-{{ .Branch }}
+          paths:
+            - "python_installer"
+
+      - run:
+          name: build MSI package.
+          command: |
+             .\win32\build.ps1 c:\Python37
+              mkdir package
+              mv .\ScalyrAgentInstaller-*.msi package\ScalyrAgentInstaller.msi
+
+      - persist_to_workspace:
+          root: package
+          paths: ScalyrAgentInstaller.msi
+
+      - store_artifacts:
+          path: package
+
+  build-linux-packages:
+    description: "Build deb and rpm packages."
+    working_directory: ~/scalyr-agent-2
+    docker:
+        - image: circleci/python:3.6-jessie
+    steps:
+      - checkout
+      - restore_cache:
+          key: deps2-tox-{{ .Branch }}-3.6-venv-{{ checksum "dev-requirements.txt" }}
+
+      - run:
+          command: |
+            sudo apt-key update
+            sudo apt update
+            sudo apt install -y ruby ruby-dev rubygems rpm build-essential --force-yes
+            sudo gem install --no-document fpm
+
+      - run:
+          command: |
+            mkdir artifacts
+            mkdir package
+
+            ls
+            pushd artifacts
+
+            python ../build_package.py deb
+            python ../build_package.py rpm
+
+            popd
+
+            cp artifacts/scalyr-agent-2*.deb package/scalyr-agent-2.deb
+            cp artifacts/scalyr-agent-2*.rpm package/scalyr-agent-2.rpm
+
+
+
+      - persist_to_workspace:
+          root: package
+          paths:
+            - scalyr-agent-2.deb
+            - scalyr-agent-2.rpm
+
+
+
+  ami-tests:
+    description: "Run scalyr agent packages sanity tests based on EC2/Libcloud."
+    working_directory: ~/scalyr-agent-2
+    docker:
+      - image: circleci/python:3.6-jessie
+    steps:
+      - add_ssh_keys:
+            fingerprints:
+              - "7b:26:91:6b:a0:9d:31:50:4c:8d:31:a3:a6:b8:f2:7a"
+      - checkout
+      - attach_workspace:
+          at: /tmp/workspace
+      - run:
+          name: run package tests in EC2 instances.
+          command: |
+            export ACCESS_KEY=${AWS_ACCESS_KEY}
+            export SECRET_KEY=${AWS_SECRET_KEY}
+            export KEY_NAME=circleci
+            export PRIVATE_KEY_PATH="~/.ssh/id_rsa_7b26916ba09d31504c8d31a3a6b8f27a"
+            python -m pip install --user -r tests/ami/requirements.txt
+            export PYTHONPATH="."
+            mkdir -p outputs
+
+            # run sanity test for each image concurrently in backgroung.
+            python tests/ami/packages_sanity_tests.py --distro=WindowsServer2012 --type=install --to-version=/tmp/workspace/ScalyrAgentInstaller.msi &> outputs/WindowsServer2012.log & \
```

Comment:
Do we have any timing information - aka how long this step takes?

I only see failed builds here so not sure - https://app.circleci.com/pipelines/github/scalyr/scalyr-agent-2?branch=windows-package-to-pyinstaller.

---

On that note, we could probably run tests for different distros in parallel since they are independent.

This would probably save quite some time (IIRC, Linux ones take around - 80 -120 seconds per distro and I imagine Windows one take quite a bit longer).